Wabbajack.Lib/NexusApi/NexusApi.cs

```diff
@@ -50,15 +51,19 @@ public UserStatus UserStatus
             }
         }
 
-        public bool IsPremium => IsAuthenticated && UserStatus.is_premium;
-
-        public string Username => UserStatus?.name;
+        public async Task<bool> IsPremium()
+        {
+            return IsAuthenticated && (await UserStatus).is_premium;
+        }
 
+        public async Task<string> Username() => (await UserStatus).name;
 
```

Comment:
The NexusApi required some discretionary choices, as the ctor and these properties were doing async stuff.  The constructor definitely needs to be a factory instead, but we could opt to offer these properties without tasks, if we want to do them at the time of the factory, rather than lazy retrieval upon access.